subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2148,37 +2273,60 @@ static int handle_request(struct zoap_packet *request,
 		return -EINVAL;
 	}
 
-	if (r == 0) {
-		/* TODO: Handle blockwise 1 */
+	if (r) {
```

Comment:
I like the restructuring here where we move the handling of the zoap_header_set_code() calls.  However, in the future, you could break this out as a small patch prior to "net: lwm2m: add firmware push support".  The code movement patches are really easy to review, but complicate reviewing larger patches.